src/server/server.rs

```diff
@@ -344,6 +396,35 @@ where
         Ok(())
     }
 
+    pub fn pause(&mut self) -> Result<()> {
+        info!("pausing the grpc server");
+
+        // Prepare the builder for resume grpc server
+        // If the builder cannot be created, then pause will be skipped;
+        let builder = self.builder_factory.create_builder(self.env.clone())?;
+        let builder = Either::Left(builder);
+        if let Some(Either::Right(server)) = self.builder_or_server.take() {
+            drop(server);
+        }
+        self.health_service
+            .set_serving_status("", ServingStatus::NotServing);
+        self.builder_or_server = Some(builder);
```

Comment:
Why not create builder on `resume()`?

Reply:
As annotations marked in `pause()`, we wanna make the preparing work completed when successfully `pause`.